MRN-Code/ansible-ops
# Feature/webpack dev server

#### asana
https://app.asana.com/0/10663154302695/39577584961564

# problem
the build process is slow.  the build process is often forgotten to be triggered when changing modes/branches

# solution
in development, host a dedicated server to serve built assets.  these assets are served directly from mem after first read.  changes done spawn a full rebuilt. rather, incremental builds occur, so, the assets are rebuilt quickly.  further, we activate live-reload, in which case, periodically, the developers browser receives the new content automatically.

# test
- merge all feature branches
- ansible deploy the latest of the affected roles (phpserver, nodejssever) to a devcoin. NOTE, that the target devcoin better be running 14.04 so that apache config updates take successfully!
- observe that COINS loads OK (assuming it did prior to deploy)
- observe that some JS assets come from an express server on port `22222`
- observe that logs are output to `/coins/www/logs/webpack-dev-server.log`
- observe that the service behaves like a system service. `sudo service wepback-dev-server restart`
- observe that changing the config file reboots the service
    - edit `/cc/js/browserApp/webpack-dev-server.js` or `webpack.config.js`. this triggeres a server reboot, COOL!@.  check out the logs and observe that it shows in the log, `/coins/www/logs/webpack-dev-server.log`
- change something that is built. `js/browser-app/global-utils.js` for instance. see if it live updates.  at the least, refresh and see that the change happens at minimum on refresh
